src/main/java/org/powerflows/dmn/engine/evaluator/entry/expression/provider/script/DefaultScriptEngineProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2018-present PowerFlows.org - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.powerflows.dmn.engine.evaluator.entry.expression.provider.script;
+
+import org.powerflows.dmn.engine.model.decision.expression.ExpressionType;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+
+public class DefaultScriptEngineProvider implements ScriptEngineProvider {
+
+    private final ScriptEngineManager scriptEngineManager;
+
+    public DefaultScriptEngineProvider(final ScriptEngineManager scriptEngineManager) {
+        this.scriptEngineManager = scriptEngineManager;
+    }
+
+    @Override
+    public ScriptEngine getScriptEngine(final ExpressionType expressionType) {
+        if (expressionType == ExpressionType.GROOVY) {
+            return scriptEngineManager.getEngineByName(expressionType.getName());
```

Comment:
this is not very generic. Instead of name field on the enum I'd have some mapping of ExpressionType to Engine...
Maybe something like EngineSource, implemented by ScriptEngineSource supporting ExpressionType GROOVY...